workspaces/liferay-sample-workspace/client-extensions/liferay-sample-etc-node/util/liferay-oauth2-resource-server.js

```diff
@@ -0,0 +1,101 @@
+'use strict';
+
+import config from './configTreePath.js';
+import cors from 'cors';
+import fetch from 'node-fetch';
+import {verify} from 'jsonwebtoken';
+import jwktopem from 'jwk-to-pem';
+import log from './log.js';
+
+const domains = config['com.liferay.lxc.dxp.domains'];
+const externalReferenceCode =
+	config['liferay.oauth.application.external.reference.codes'].split(',')[0];
+const lxcDXPMainDomain = config['com.liferay.lxc.dxp.mainDomain'];
+
+const lxcDXPServerProtocol = config['com.liferay.lxc.dxp.server.protocol'];
+const oauth2JWKSURI =
+	lxcDXPServerProtocol +
+	'://' +
+	lxcDXPMainDomain +
+	(config[externalReferenceCode + '.oauth2.jwks.uri'] || '/o/oauth2/jwks');
+const allowList = domains
+	.split(',')
+	.map((domain) => lxcDXPServerProtocol + '://' + domain);
+
+const corsOptions = {
+	origin: function (origin, callback) {
+		if (allowList.includes(origin)) {
+			callback(null, true);
+		}
+		else {
+			callback(null, false);
+		}
+	},
+};
+
+export async function corsWithReady(req, res, next) {
+	if (req.originalUrl === config.readyPath) {
+		return next();
+	}
+	return cors(corsOptions)(req, res, next);
+}
+
+export async function liferayJWT(req, res, next) {
+	if (req.path === config.readyPath) {
+		return next();
+	}
+
+	const authorization = req.headers.authorization;
+
+	if (!authorization) {
+		res.status(401).send('No authorization header');
+		return;
+	}
+
+	const bearerToken = req.headers.authorization.split('Bearer ')[1];
+
+	try {
+		const jwksResponse = await fetch(oauth2JWKSURI);
+		if (jwksResponse.status === 200) {
+			const jwks = await jwksResponse.json();
+			const jwksPublicKey = jwktopem(jwks.keys[0]);
+			const decoded = verify(bearerToken, jwksPublicKey, {
```

Comment:
This goes for all the js files here, but we make sure there is a newline between the declaration of a variable and its use.

So for this chunk, it should be `jwksPublicKey` declaration and then followed by new line and then using it in `decoded`